ui/apps/platform/src/Components/CompoundSearchFilter/components/CompoundSearchFilter.tsx

```diff
@@ -80,51 +86,26 @@ function CompoundSearchFilter({
                 }}
                 config={config}
             />
-            <AttributeSelector
-                selectedEntity={selectedEntity}
-                selectedAttribute={selectedAttribute}
-                onChange={(value) => {
-                    setSelectedAttribute(value as SearchFilterAttributeName);
-                    setInputValue('');
-                }}
-                config={config}
-            />
+            {hasMultipleAttributes && (
+                <AttributeSelector
+                    selectedEntity={selectedEntity}
+                    selectedAttribute={selectedAttribute}
+                    onChange={(value) => {
+                        setSelectedAttribute(value as SearchFilterAttributeName);
+                        setInputValue('');
+                    }}
+                    config={config}
+                />
+            )}
```

Comment:
Hmm this is a good question - do we have a use case where this happens so far?

I could definitely see it being confusing if there was an unlabeled date picker that was rendered in the search bar.

Instead of hiding the dropdown, could we make it read-only, or make the attribute name a label instead of an input element?

Reply:
A disabled/read-only version is also possible. I can see how that looks. If it doesn't look bad, maybe we could do that

Reply:
Here is a screenshot of what it looks like when it's disabled/read-only vs. unrendered (above screenshot).

![Screenshot 2024-06-05 at 2 03 35 PM](https://github.com/stackrox/stackrox/assets/4805485/424cac5e-96cd-4532-b35f-101b5a1b93e8)

Reply:
Hmm honestly I'm not sure I have a strong opinion on this one. If it is easiest, it might be just as well to keep the full menu item without the disabled styles. A user that opens the menu will see there is only one option anyway 🤷‍♂️ 

Unless the team or UX feels otherwise I think whatever the simplest solution is will be fine.